extension/k8sleaderelector/README.md

```diff
@@ -16,12 +16,13 @@ This extension enables OpenTelemetry components to run in HA mode across a Kuber
 ## How It Works
 
 The extension uses k8s.io/client-go/tools/leaderelection to perform leader election. The component that owns the lease becomes the leader and runs the function defined in onStartedLeading. If the leader loses the lease, it runs the function defined in onStoppedLeading, stops its operation, and waits to acquire the lease again.
+> When the pod is deleted then the lease object is still left in the cluster. This needs to be deleted manually.
```

Comment:
This statement is a bit unclear. Does it mean that if a leader pod is kicked, there won't be any leader?

Reply:
I wanted to highlight here that when eg someone removes the otel collector deployment which is using the k8s leader elector, then the lease object is still left behind. This needs to be cleared manually.

Reply:
added instruction